src/snabbdom.ts

```diff
@@ -57,10 +57,29 @@ export function init(modules: Array<Partial<Module>>, domApi?: DOMAPI) {
     }
   }
 
-  function emptyNodeAt(elm: Element) {
+  function toVNode(elm: Element): VNode {
     const id = elm.id ? '#' + elm.id : '';
-    const c = elm.className ? '.' + elm.className.split(' ').join('.') : '';
-    return vnode(api.tagName(elm).toLowerCase() + id + c, {}, [], undefined, elm);
+    const cn = elm.getAttribute('class');
+    const c = cn ? '.' + cn.split(' ').join('.') : '';
+    const sel = api.tagName(elm).toLowerCase() + id + c;
+    const attrs: any = {};
+    for (let i = 0, elmAttrs = elm.attributes, n = elmAttrs.length; i < n; i++) {
+      attrs[elmAttrs[i].nodeName] = elmAttrs[i].nodeValue;
+    }
+    delete attrs.id;
+    delete attrs.class;
+    const children: Array<VNode> = [];
+    for (let i = 0, elmChildren = elm.childNodes, n = elmChildren.length; i < n; i++) {
+      const node = elmChildren[i];
+      if (api.isElement(node)) {
+        children.push(toVNode(node));
+      } else if (api.isText(node)) {
+        children.push(
+          (vnode as any)(undefined, undefined, undefined, node.textContent)
```

Comment:
I'd also put a `getTextContent` function in the API, in conformity with `setTextContent`.